common/common.go

```diff
@@ -57,3 +44,17 @@ func sliceSuffix(name string) (string, error) {
 	}
 	return found[0], nil
 }
+
+// UnitBase returns the base of the unit name. The base is considered
+// everything up to the @ character, if any, or the given name without
+// extension.
+//
+//   app@1.service  =>  app
```

Comment:
It might be an idea to use the "example functions" available in https://golang.org/pkg/testing/